File: packages/backend/src/features/answer/answer.model.ts
import { Column, Entity, ManyToOne } from 'typeorm';
import { QuizzemModel } from '~/src/core/domain/quizzem.model';
import { QuestionModel } from '~/src/features/question/question.model';

@Entity()
export class AnswerModel extends QuizzemModel {
  @Column()
  text: string = '';

  @ManyToOne(() => QuestionModel, (question) => question.answers)
  question!: QuestionModel;
}

File: packages/backend/src/main/kotlin/quizzem/backend/features/answer/model/AnswerModel.kt
package quizzem.backend.features.answer.model

import jakarta.persistence.*
import quizzem.backend.features.question.model.QuestionModel
import java.time.Instant
import java.util.*

@Entity
@Table(name = "answers")
class AnswerModel(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: UUID? = null,

    var createdAt: Instant = Instant.now(),

    var updatedAt: Instant = createdAt,

    @Column(length = 400) var text: String = "",

    @ManyToOne @JoinColumn(name = "question_id") var question: QuestionModel
) {
    @PreUpdate
    fun onUpdate() {
        updatedAt = Instant.now()
    }
}
